test/e2e/copy_from_image_test.go

```diff
@@ -210,32 +174,35 @@ func TestCopyToTarWithoutNonDistributableLayersFlagButContainsANonDistributableL
 		// copy to tar
 		imgpkg.Run([]string{"copy", "-i", env.RelocationRepo, "--to-tar", tarFilePath, "--include-non-distributable"})
 
-		var stdOutWriter bytes.Buffer
-		imgpkg.RunWithOpts([]string{"copy", "--tar", tarFilePath, "--to-repo", repoToCopyName}, RunOpts{
-			StdoutWriter: &stdOutWriter,
-			StderrWriter: &stdOutWriter,
+		stderr := bytes.NewBufferString("")
+		imgpkg.RunWithOpts([]string{"copy", "--tar", tarFilePath, "--to-repo", repoToCopyName, "--include-non-distributable"}, RunOpts{
+			StderrWriter: stderr,
 		})
 
-		if !strings.Contains(stdOutWriter.String(), "Skipped layer due to it being non-distributable.") {
-			t.Fatalf("Expected warning message to user, specifying which layer was skipped. But found: %s", stdOutWriter.String())
-		}
-		digestOfNonDistributableLayer, err := name.NewDigest(repoToCopyName + "@" + nonDistributableLayerDigest)
+		digest, err := name.NewDigest(repoToCopyName + "@" + nonDistributableLayerDigest)
 		if err != nil {
 			t.Fatalf("Unable to determine the digest of the non-distributable layer. Got: %v", err)
 		}
 
-		layer, err := remote.Layer(digestOfNonDistributableLayer, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+		layer, err := remote.Layer(digest, remote.WithAuthFromKeychain(authn.DefaultKeychain))
 		if err != nil {
 			t.Fatalf("Unable to fetch the layer of the copied image: %v", err)
 		}
 
 		_, err = layer.Compressed()
-		if err == nil {
-			t.Fatalf("Expected non-distributable layer to NOT be copied into registry, however it was")
+		if err != nil {
+			t.Fatalf("Expected to find a non-distributable layer however it wasn't found. Got response code: %v", err)
+		}
+
+		imageDigest := fmt.Sprintf("@%s", extractDigest(t, stderr.String()))
+		cmd := exec.Command("docker", "pull", repoToCopyName+imageDigest)
```

Comment:
ideally we wouldnt have any dependence on docker. we can use imgpkg pull -i here.

Reply:
I can make this change, but i'm confused to why it is ideal.